### PR TITLE
chore: apply nursery and pedantic clippy lints

### PR DESCRIPTION
**Pull request by aleksuss**
## Description

The PR just increase clippy lints up to the nursery and pedantic in main sub-crates.

### CODE REVIEW

File: engine-types/src/types/mod.rs [hunk 1]
```diff
@@ -128,11 +131,14 @@ impl<T> Stack<T> {
         self.stack.push(value);
     }
 
+    #[must_use]
+    #[allow(clippy::missing_const_for_fn)]
```
Comment (birchmd) [inline engine-types/src/types/mod.rs:135]: Why allow?

File: engine/src/lib.rs [hunk 2]
```diff
@@ -196,7 +202,7 @@ mod contract {
     /// to make any necessary changes to the state such that it aligns with the newly deployed
     /// code.
     #[no_mangle]
-    pub extern "C" fn state_migration() {
+    pub const extern "C" fn state_migration() {
```
Comment (birchmd) [inline engine/src/lib.rs:205]: Similarly here, I think it makes sense to not mark as `const` since a non-trivial implementation would not be `const`.

**Review by birchmd — COMMENTED** [review]
Thanks for going through all of this! It looks like it was a lot of work!

File: engine-tests/src/tests/mod.rs [hunk 3]
```diff
@@ -16,10 +16,12 @@ mod prepaid_gas_precompile;
...
 mod self_destruct_state;
 mod standalone;
 mod standard_precompiles;
 mod state_migration;
-pub(crate) mod uniswap;
-pub(crate) mod xcc;
+pub mod uniswap;
+pub mod xcc;
+
+const ZERO_ADDRESS_FIX_HEIGHT: u64 = 61_200_152;
```
[the remark this answers is not in the document]
Comment (aleksuss) [reply engine-tests/src/tests/mod.rs:27]: Yes. You're right. `--all-features` checks the tests where that constant is used.

File: engine/src/lib.rs (hunk 2 above)
Comment (aleksuss) [reply engine/src/lib.rs:205]: Yes. Makes sense.

File: engine-precompiles/src/account_ids.rs [hunk 4]
```diff
@@ -21,16 +22,16 @@ pub struct PredecessorAccount<'a, E> {
...
     use aurora_engine_types::types::Address;
 
-    /// predecessor_account_id precompile address
+    /// `predecessor_account_id` precompile address
     ///
     /// Address: `0x723ffbaba940e75e7bf5f6d61dcbf8d9a4de0fd7`
     /// This address is computed as: `&keccak("predecessorAccountId")[12..]`
     pub const ADDRESS: Address =
-        crate::make_address(0x723ffbab, 0xa940e75e7bf5f6d61dcbf8d9a4de0fd7);
+        crate::make_address(0x723f_fbab, 0xa940_e75e_7bf5_f6d6_1dcb_f8d9_a4de_0fd7);
```
[the remark this answers is not in the document]
Comment (aleksuss) [reply engine-precompiles/src/account_ids.rs:30]: To be honest, I don't like this as well. 😄

File: engine-standalone-tracing/src/types/mod.rs [hunk 5]
```diff
@@ -105,7 +115,9 @@ impl std::iter::FromIterator<[u8; 32]> for LogStack {
 pub struct ProgramCounter(pub u32);
 
 impl ProgramCounter {
-    /// Performs the conversion into a u32.
+    /// Performs the conversion into u32.
+    #[must_use]
+    #[allow(clippy::missing_const_for_fn)]
```
[the remark this answers is not in the document]
Comment (aleksuss) [reply engine-standalone-tracing/src/types/mod.rs:120]: Good catch.

File: engine-types/src/types/mod.rs (hunk 1 above)
Comment (aleksuss) [reply engine-types/src/types/mod.rs:135]: And here seems a bug in the clippy. There a destructor of the `self` will be invoked which makes it impossible to make this method `const`.

File: engine-precompiles/src/alt_bn256.rs [hunk 6]
```diff
@@ -48,18 +48,21 @@ mod consts {
     /// Pair element length.
     pub(super) const PAIR_ELEMENT_LEN: usize = 192;
 
+    #[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (aleksuss) [reply engine-precompiles/src/alt_bn256.rs:51]: I couldn't find usage of this constant regardless of a feature.

**Comment by aleksuss** [comment]
@birchmd thank you for reviewing.

**Comment by joshuajbouw** [comment]
Lets hold off on this until our main goals for this quarter are complete.

**Comment by joshuajbouw** [comment]
Lets continue on this.

**Review by joshuajbouw — APPROVED** [review]

**Review by mrLSD — APPROVED** [review]

File: engine-precompiles/src/alt_bn256.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (birchmd) [reply engine-precompiles/src/alt_bn256.rs:51]: If it's legitimately not used should we delete it instead of allowing the dead code?

**Review by birchmd — APPROVED** [review]

File: engine-precompiles/src/alt_bn256.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (aleksuss) [reply engine-precompiles/src/alt_bn256.rs:51]: Ok. I will delete this. I decided to leave it for use in the future.

**Pull request merged**